src/main/java/com/diffplug/gradle/oomph/ConventionJdt.java

```diff
@@ -33,24 +39,33 @@
  * ```gradle
  * oomphIde {
  *     jdt {
+ *         
  *         installedJre {
  *             version = '1.6.0_45'
  *             installedLocation = new File('C:/jdk1.6.0_45')
  *             markDefault = true // or false
  *             executionEnvironments = ['JavaSE-1.6'] // any execution environments can be specified here.
  *         }
+ *         compilerComplianceLevel('1.6')
+ *         classpathVariable('myClasspath', '/var/lib/repo')
  *     }
  * }
  * ```
  */
 public class ConventionJdt extends OomphConvention {
+	final static String JDT_CORE_PREFS = ".metadata/.plugins/org.eclipse.core.runtime/.settings/org.eclipse.jdt.core.prefs";
+	final static List<String> JDT_COMPLIANCE_PROPS = Collections.unmodifiableList(Arrays.asList("org.eclipse.jdt.core.compiler.codegen.targetPlatform", "org.eclipse.jdt.core.compiler.compliance", "org.eclipse.jdt.core.compiler.source"));
```

Comment:
ImmutableList was the right way to go.  I pushed the change in.

You're free to do your worst now. :)